packages/client/src/usage.ts

```diff
@@ -0,0 +1,51 @@
+import { ServiceEndpoint } from './configuration';
+import makeRequest from './makeRequest';
+
+interface UsageDescription {
+  readonly usage: number;
+  readonly limit: number;
+}
+
+interface ErrorResponse {
+  statusCode: number;
+  message: string;
+}
+
+export interface UsageReport {
+  readonly events: UsageDescription;
+  readonly cycleStart: Date;
+  readonly cycleEnd: Date;
+}
+
+export default class Usage {
+  private static readonly usagePath = 'v1/orgs/usage';
+
+  static async get(orgId?: number): Promise<UsageReport> {
+    const response = await makeRequest({
+      service: ServiceEndpoint.ServiceApi,
+      path: this.usagePath,
+      query: orgId !== undefined ? [['orgId', String(orgId)]] : undefined,
+    });
+
+    const responseJson = JSON.parse(response.body.toString('utf-8')) as UsageReport | ErrorResponse;
+    if (!response.ok) {
+      if ('message' in responseJson) throw new Error(responseJson.message);
+      throw new Error(`Got unexpected status code ${response.statusCode}`);
+    }
+
+    return responseJson as UsageReport;
+  }
+
+  static async update(events: number, orgId?: number): Promise<void> {
```

Comment:
I don’t follow org id here. The user may belong to multiple orgs. What we do know is the app name as reported by appmap.yml. Also the account identified by the api key.